apps/server/src/shared/domain/entity/coursegroup.entity.spec.ts

```diff
@@ -31,16 +32,47 @@ describe('CourseEntity', () => {
 		});
 	});
 
-	// TODO check if and how we need this
-	describe('getParent', () => {
-		it('should return the right id.', () => {
-			const name = 'someName';
-			const course = courseFactory.build();
-			const courseGroup = new CourseGroup({ name, course });
+	describe('getStudentIds is called', () => {
+		describe('when students exist', () => {
+			const setup = () => {
+				const student1 = userFactory.buildWithId();
+				const student2 = userFactory.buildWithId();
+				const student3 = userFactory.buildWithId();
+				const students = [student1, student2, student3];
+				const studentIds = [student1.id, student2.id, student3.id];
+
+				const courseGroup = courseGroupFactory.build({ students });
+
+				return { courseGroup, studentIds };
+			};
+
+			it('should return the userIds of the students', () => {
+				const { courseGroup, studentIds } = setup();
+
+				const result = courseGroup.getStudentIds();
+
+				expect(result.length).toEqual(3);
+				expect(result.includes(studentIds[0])).toBe(true);
+				expect(result.includes(studentIds[1])).toBe(true);
+				expect(result.includes(studentIds[2])).toBe(true);
+			});
+		});
+
+		describe('when course group is not populated', () => {
+			const setup = () => {
+				const courseGroup = courseGroupFactory.build();
+				Object.assign(courseGroup, { students: undefined });
```

Comment:
Build is added a empty collection. But with mikroORM we can run into the situation that coursegroup has only .id and _id without any other key (is not populated, like submission.coursegroup without populate of coursegroup). But typescript say it is not possible. This is also the reason why i must use Object.assign otherwise ts post an error.